samples/performance/constant_data/constant_data_tutorial.md

```diff
@@ -0,0 +1,403 @@
+<!--
+- Copyright (c) 2020, Arm Limited and Contributors
+-
+- SPDX-License-Identifier: Apache-2.0
+-
+- Licensed under the Apache License, Version 2.0 the "License";
+- you may not use this file except in compliance with the License.
+- You may obtain a copy of the License at
+-
+-     http://www.apache.org/licenses/LICENSE-2.0
+-
+- Unless required by applicable law or agreed to in writing, software
+- distributed under the License is distributed on an "AS IS" BASIS,
+- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+- See the License for the specific language governing permissions and
+- limitations under the License.
+-
+-->
+
+# **Constant data in Vulkan**
+
+## **Overview**
+
+The Vulkan API exposes a few different ways in which we can send uniform data into our shaders. There are enough methods that it raises the question "Which one is fastest?", and more often than not the answer is "It depends".
+
+This sample aims to break this question down by highlighting the different methods we can use, and the performance implications that each of them bring. Then you, the developer, can pick the right one.
+
+## **Contents**
+
+- [What is Constant Data?](#what-is-constant-data)
+    - [Introduction](#introduction)
+    - [Constant data in Vulkan shaders](#constant-data-in-vulkan-shaders)
+    - [Vulkan API](#vulkan-api)
+- [Sample Overview](#sample-overview)
+- [Push Constants](#push-constants)
+- [Descriptor Sets](#descriptor-sets)
+- [Dynamic Descriptor Sets](#dynamic-descriptor-sets)
+- [Update-after-bind Descriptor Sets](#update-after-bind-descriptor-sets)
+- [Buffer Object Arrays](#buffer-object-arrays)
+- [Further reading](#further-reading)
+- [Best practice summary](#best-practice-summary)
+
+## **What is constant data?**
+
+### **Introduction**
+
+Constant data is a form of information that is supplied to the graphics pipeline to help with shader computations.
+
+In theory, this data can be anything we want it to be, for instance it can be used for things such as calculating where an object should be placed inside our world, or computing the overall brightness of an object based on the lights in the scene.
+
+It differs from other data (e.g. input vertex data) in the sense that it remains *constant* across every shader invocation of a draw call.
```

Comment:
input vertex data is constant though, but it is not uniform. I feel there is a blending of `constant` data  and `uniform` data here